client/core/exchangeratefetcher.go

```diff
@@ -196,7 +199,7 @@ func fetchMessariRates(ctx context.Context, log dex.Logger, assets map[uint32]*S
 		slug := strings.ToLower(asset.Symbol)
 		reqStr := fmt.Sprintf(messariURL, slug)
 
-		if err := getInto(ctx, reqStr, res); err != nil {
+		if err := getRates(ctx, reqStr, res); err != nil {
```

Comment:
The reason we think we're getting zero from Messari is this:

```go
		res := new(struct {
			Data struct {
				MarketData struct {
					Price float64 `json:"price_usd"`
				} `json:"market_data"`
			} `json:"data"`
		})
```

When we unmarshal some other JSON into that, it just leaves that `data.market_data.price_usd` as zero, but it's not an unmarshal error, so we end up putting that in `fiatRates[assetID] = res.Data.MarketData.Price` below.

Reply:
> We should only try to unmarshal if the response code was 200 (http.StatusOK).

I agree, but I will have to run Messari for a longer period and see what other `resp.body` we are receiving.

Reply:
The response is here: https://github.com/decred/dcrdex/pull/1974#issuecomment-1330782812

But only the status code matters